Brokerages/DefaultOrderBook.cs

```diff
@@ -153,21 +161,33 @@ public void RemoveBidRow(decimal price)
         /// <param name="price">The ask price level to be removed</param>
         public void RemoveAskRow(decimal price)
         {
-            lock(_locker)
+            lock (_locker)
             {
-                _asks.Remove(price);
+                Asks.Remove(price);
             }
 
             if (price == BestAskPrice)
             {
-                lock(_locker)
+                lock (_locker)
                 {
-                    BestAskPrice = _asks.Keys.FirstOrDefault();
-                    BestAskSize = BestAskPrice > 0 ? _asks[BestAskPrice] : 0;
+                    BestAskPrice = Asks.Keys.FirstOrDefault();
+                    BestAskSize = BestAskPrice > 0 ? Asks[BestAskPrice] : 0;
                 }
 
                 BestBidAskUpdated?.Invoke(this, new BestBidAskUpdatedEventArgs(_symbol, BestBidPrice, BestBidSize, BestAskPrice, BestAskSize));
             }
         }
+
+        /// <summary>
+        /// Common price level removal method
+        /// </summary>
+        /// <param name="priceLevel"></param>
+        public void RemovePriceLevel(decimal priceLevel)
+        {
+            if (Asks.ContainsKey(priceLevel))
+                RemoveAskRow(priceLevel);
+            else if (Bids.ContainsKey(priceLevel))
+                RemoveBidRow(priceLevel);
+        }
```

Comment:
Upon further review, I think we should `lock` the two `ContainsKey` calls, so access to the two dictionaries is synchronized everywhere.